robottelo/hosts.py

```diff
@@ -1674,14 +1675,31 @@ def get_caller():
                 if frame.function.startswith('test_'):
                     return frame.function
 
-        return Session(
-            session_name=testname or get_caller(),
-            user=user or settings.server.admin_username,
-            password=password or settings.server.admin_password,
+        test_name = testname or get_caller()
+
+        if getattr(self, 'organization') and getattr(self, 'location') and user is None:
+            login = f'{test_name}_{gen_string("alphanumeric")}'
```

Comment:
you are overriding the `login=True`, try to use something different keyword e.g. `user_name`

Reply:
Good catch, will update